numpy/random/generator.pyx

```diff
@@ -3475,21 +3492,41 @@ cdef class Generator:
 
         # GH10839, ensure double to make tol meaningful
         cov = cov.astype(np.double)
-        (u, s, v) = svd(cov)
+        if method == 'svd':
+            from numpy.dual import svd
+            (u, s, vh) = svd(cov)
+        elif method == 'eigh':
+            from numpy.dual import eigh
+            (s, u)  = eigh(cov)
+        else:
+            from numpy.dual import cholesky
+            l = cholesky(cov)
 
-        if check_valid != 'ignore':
+        # make sure check_valid is ignored whe method == 'cholesky'
+        # since the decomposition will have failed if cov is not valid.
+        if check_valid != 'ignore' and method != 'cholesky':
             if check_valid != 'warn' and check_valid != 'raise':
-                raise ValueError("check_valid must equal 'warn', 'raise', or 'ignore'")
-
-            psd = np.allclose(np.dot(v.T * s, v), cov, rtol=tol, atol=tol)
+                raise ValueError(
+                    "check_valid must equal 'warn', 'raise', or 'ignore'")
+            if method == 'svd':
+                psd = np.allclose(np.dot(vh.T * s, vh), cov, rtol=tol, atol=tol)
+            else:
+                psd = not np.any(s < -tol)
```

Comment:
Should there be a check that `s >= 0` for the `eigh` path here?

I'm basing this remark off these lines, which suggest that eigh needs a little massaging to be a drop-in for svd:

https://github.com/numpy/numpy/blob/2be03c8d25b14b654064e953feac7d210e6bd44d/numpy/linalg/linalg.py#L1594-L1604

For a similar reason, it might be worth reversing `u` and `s`, so that the factor comes out identical to the svd case, rather than flipped. Maybe it makes no difference.

Reply:
There were concerns previously about eigenvalues possibly ending up having a negative sign for positive semi-definite matrices due to numerical error and it was suggested on this PR that a check for s>=0 should be explicitly added. Also I think `u` and `s` might be reversed already inside the code that generates them because adding a `[..., ::-1]` for them in the code causes tests to fail and the function to return the wrong values. So I think there might not be a need to do it. See:
<details><summary>ERROR</summary>
<p>
```text
elf = <numpy.random.tests.test_generator_mt19937.TestRandomDist object at 0x7fc1338b8a58>, method = 'eigh'

    @pytest.mark.parametrize("method", ["svd", "eigh", "cholesky"])
    def test_multivariate_normal(self, method):
        random = Generator(MT19937(self.seed))
        mean = (.123456789, 10)
        cov = [[1, 0], [0, 1]]
        size = (3, 2)
        actual = random.multivariate_normal(mean, cov, size, method=method)
        desired = np.array([[[-1.747478062846581,  11.25613495182354  ],
                             [-0.9967333370066214, 10.342002097029821 ]],
                            [[ 0.7850019631242964, 11.181113712443013 ],
                             [ 0.8901349653255224,  8.873825399642492 ]],
                            [[ 0.7130260107430003,  9.551628690083056 ],
                             [ 0.7127098726541128, 11.991709234143173 ]]])
    
>       assert_array_almost_equal(actual, desired, decimal=15)
E       AssertionError: 
E       Arrays are not almost equal to 15 decimals
E       
E       Mismatch: 100%
E       Max absolute difference: 3.1270698
E       Max relative difference: 2.1264784
E        x: array([[[ 1.37959174082354 ,  8.129065148153419],
E               [ 0.465458886029821,  8.879809873993379]],
E       ...
E        y: array([[[-1.747478062846581, 11.25613495182354 ],
E               [-0.996733337006621, 10.342002097029821]],
E       ...

actual     = array([[[ 1.37959174,  8.12906515],
        [ 0.46545889,  8.87980987]],

       [[ 1.3045705 , 10.66154517],
        [-1.00271781, 10.76667818]],

       [[-0.32491452, 10.58956922],
        [ 2.11516602, 10.58925308]]])
cov        = [[1, 0], [0, 1]]
desired    = array([[[-1.74747806, 11.25613495],
        [-0.99673334, 10.3420021 ]],

       [[ 0.78500196, 11.18111371],
        [ 0.89013497,  8.8738254 ]],

       [[ 0.71302601,  9.55162869],
        [ 0.71270987, 11.99170923]]])
mean       = (0.123456789, 10)
method     = 'eigh'
random     = Generator(MT19937) at 0x7FC133E81E58
self       = <numpy.random.tests.test_generator_mt19937.TestRandomDist object at 0x7fc1338b8a58>
size       = (3, 2)

```
</p>
</details>

Reply:
Apologies, I'm being stupid about the negative eigenvalues, didn't see the very line I commented on here! Will investigate the reversion, that's very strange